lib/backend/syncersv1/updateprocessors/rules.go

```diff
@@ -46,6 +49,45 @@ func RuleAPIV2ToBackend(ar apiv2.Rule) model.Rule {
 		notICMPCode = ar.NotICMP.Code
 		notICMPType = ar.NotICMP.Type
 	}
+	
+	// If we have any selector specified, then we may need to add the namespace selector.
+	// We do this if this policy is namespaced AND if the Selector does not have any other
+	// k8s namespace (profile label) selector in it.
+	// TODO this is TEMPORARY CODE:  We currently do a simple regex to search for kns. in the
+	// selector to see if we are performing k8s namespace queries.
+	nsSelector := fmt.Sprintf("%s == '%s'", apiv2.LabelNamespace, ns)
+	if ns != "" && (ar.Source.Selector != "" || ar.Source.NotSelector != "") {
+		logCxt := log.WithFields(log.Fields{
+			"Namespace": ns,
+			"Selector": ar.Source.Selector,
+			"NotSelector": ar.Source.NotSelector,
+		})
+		logCxt.Debug("Maybe update source Selector to include namespace")
+		if !strings.Contains(ar.Source.Selector, "kns.") {
+			logCxt.Debug("Updating source selector")
+			if ar.Source.Selector == "" {
```

Comment:
At this point we've already identified that `ar.Source.Selector != ""` so we can remove this check.

Reply:
no we haven't

Reply:
first check is if either are != ""

And the ".kns" check is a negated check

Reply:
Yeah you're right. Carry on.